[libc] Standalone libposix lacks the stdio stream accessors (stdin/stdout/stderr) declared by <stdio.h>

## Description

The sysroot `<stdio.h>` exposes the standard streams as **accessor functions**
backed by `libc_stdio`, plus the full `FILE` API:

```c
typedef struct { ... int ungetc_buf; } FILE;
extern FILE *stdin(void);
extern FILE *stdout(void);
extern FILE *stderr(void);
#define stdin  (stdin())
#define stdout (stdout())
#define stderr (stderr())
extern size_t fwrite(const void*, size_t, size_t, FILE*);
/* ... fopen/fread/fgets/fprintf/setvbuf/... */
```

But the **standalone** `libposix.a` implements almost none of this. A symbol
dump shows it provides only `malloc`/`free`/`calloc`/`realloc` and the raw
`open`/`close`/`read`/`write`; there is **no** `stdin()`/`stdout()`/`stderr()`,
no `fopen`/`fwrite`/`fprintf`/`setvbuf`, etc.

Consequently a standalone C program that uses stdio links those symbols from the
cross toolchain's newlib (`libc.a`) — but the standard streams resolve to the
sysroot's `stdin()`/`stdout()`/`stderr()` functions, which are undefined:

```
ld: undefined reference to `stdin'
ld: undefined reference to `stdout'
ld: undefined reference to `stderr'
```

There is also a latent `FILE` type mismatch: the header's `FILE` is the Rust
`libc_stdio` struct, whereas the linked stdio routines are newlib's and operate
on newlib's `__FILE`.

## How to Reproduce

1. Build Nanvix dev (0.16.120) release/standalone; resolve the sysroot.
2. Cross-compile + link any `-nostdinc` C program that writes to `stdout`
   (`fprintf(stdout, ...)`, `setvbuf(stdout, ...)`, etc.).
3. The link fails with undefined references to `stdin`/`stdout`/`stderr`.
   `nm libposix.a` confirms these and the rest of stdio are absent.

## Expected Behavior

The standalone `libposix` should either provide the `libc_stdio` symbols its own
`<stdio.h>` advertises (so the FILE model is self-consistent), or the header
should match the libc that standalone programs actually link.

## Impact

Any standalone C port that uses buffered stdio fails to link until it supplies
its own stream plumbing.

## Workaround (in the BusyBox port)

A compat `<stdio.h>` undefines the `stdin()/stdout()/stderr()` macros and remaps
them onto newlib's reentrancy struct, whose first members are
`{ int _errno; FILE *_stdin, *_stdout, *_stderr; }`:

```c
extern struct __nanvix_reent *_impure_ptr;
#define stdin  (_impure_ptr->_stdin)
#define stdout (_impure_ptr->_stdout)
#define stderr (_impure_ptr->_stderr)
```

## Related Issues

Same root class as the `<setjmp.h>` / `libc_setjmp` mismatch: the generated
headers describe `libc_*` crates that the standalone `libposix` does not bundle.

--- a/.github/skills/user-app-development/SKILL.md
+++ b/.github/skills/user-app-development/SKILL.md
@@ -147,7 +147,7 @@ For low-level debugging, the standalone UserVM is still available:
 1. Create directory at `src/user/<name>/`.
 2. Create a `Makefile` following the pattern in existing C applications.
 3. Write source files using the Nanvix POSIX layer (`#include <nanvix/...>`).
-4. The application links against `libposix.a`, `libc.a`, and the custom linker script at
+4. The application links against `libc.a` and the custom linker script at
    `build/user/linker/x86/user.ld`.
 
 ## Logging

--- a/build/user/linker/x86/user.ld
+++ b/build/user/linker/x86/user.ld
@@ -73,7 +73,7 @@ SECTIONS
 	/*
 	 * Constructor / destructor arrays (the primary ctor/dtor mechanism).
 	 *
-	 * `__nanvix_libc_start_main` (libposix start.rs) walks `.preinit_array`
+	 * `__nanvix_libc_start_main` (nanvix_libc start.rs) walks `.preinit_array`
 	 * then `.init_array` before calling `main`, and `.fini_array` (in reverse)
 	 * after `main` returns.  The bound symbols below are provided HERE — not by
 	 * GCC crtbegin/crtend — because Nanvix ships no crti/crtn/crtbegin/crtend.

--- a/build/user/linker/x86_64/user.ld
+++ b/build/user/linker/x86_64/user.ld
@@ -55,7 +55,7 @@ SECTIONS
 	/*
 	 * Constructor / destructor arrays (the primary ctor/dtor mechanism).
 	 *
-	 * `__nanvix_libc_start_main` (libposix start.rs) walks `.preinit_array`
+	 * `__nanvix_libc_start_main` (nanvix_libc start.rs) walks `.preinit_array`
 	 * then `.init_array` before calling `main`, and `.fini_array` (in reverse)
 	 * after `main` returns.  The bound symbols below are provided HERE — not by
 	 * GCC crtbegin/crtend — because Nanvix ships no crti/crtn/crtbegin/crtend.

--- a/doc/run-linux.md
+++ b/doc/run-linux.md
@@ -62,7 +62,7 @@ boot:
 ```
 
 The three daemon binaries (`procd.elf`, `memd.elf`, `vfsd.elf`) are shipped in the release
-archive under `bin/`. Your application binary must be compiled and linked against `libposix.a`
+archive under `bin/`. Your application binary must be compiled and linked against `libc.a`
 using the `user.ld` linker script (both also in the release archive).
 
 Once the image is created, pass it to `nanvixd` as the program argument:

--- a/doc/run-windows.md
+++ b/doc/run-windows.md
@@ -54,7 +54,7 @@ boot:
 ```
 
 The three daemon binaries (`procd.elf`, `memd.elf`, `vfsd.elf`) are shipped in the release
-archive under `bin/`. Your application binary must be compiled and linked against `libposix.a`
+archive under `bin/`. Your application binary must be compiled and linked against `libc.a`
 using the `user.ld` linker script (both also in the release archive).
 
 Once the image is created, pass it to `nanvixd` as the program argument:

